app/test/prepare_proposal_context_test.go

```diff
@@ -0,0 +1,100 @@
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+// TestTimeInPrepareProposalContext checks for an edge case where the block time
+// needs to be included in the sdk.Context that is being used in the
+// antehandlers. If a time is not included in the context, then the second
+// transaction in this test will always be filtered out, result in vesting
+// accounts never being able to spend funds.
+func TestTimeInPrepareProposalContext(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping TestTimeInPrepareProposalContext test in short mode.")
+	}
+	accounts := make([]string, 35)
+	for i := 0; i < len(accounts); i++ {
+		accounts[i] = tmrand.Str(9)
+	}
+	cfg := testnode.DefaultConfig().WithAccounts(accounts)
+	cctx, _, _ := testnode.NewNetwork(t, cfg)
+	ecfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	vestAccName := "vesting"
+	type test struct {
+		name         string
+		msgFunc      func() (msgs []sdk.Msg, signer string)
+		expectedCode uint32
+	}
+	tests := []test{
+		{
+			name: "create continuous vesting account with a start time in the future",
+			msgFunc: func() (msgs []sdk.Msg, signer string) {
+				_, _, err := cctx.Keyring.NewMnemonic(vestAccName, keyring.English, "", "", hd.Secp256k1)
+				require.NoError(t, err)
+				sendAcc := accounts[0]
+				sendingAccAddr := testfactory.GetAddress(cctx.Keyring, sendAcc)
+				vestAccAddr := testfactory.GetAddress(cctx.Keyring, vestAccName)
+				msg := vestingtypes.NewMsgCreateVestingAccount(
+					sendingAccAddr,
+					vestAccAddr,
+					sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(1000000))),
+					time.Now().Unix(),
+					time.Now().Add(time.Second*100).Unix(),
+					false,
+				)
+				return []sdk.Msg{msg}, sendAcc
+			},
+			expectedCode: abci.CodeTypeOK,
+		},
+		{
+			name: "send funds from the vesting account after it has been created",
+			msgFunc: func() (msgs []sdk.Msg, signer string) {
+				sendAcc := accounts[1]
+				sendingAccAddr := testfactory.GetAddress(cctx.Keyring, sendAcc)
+				vestAccAddr := testfactory.GetAddress(cctx.Keyring, vestAccName)
+				msg := banktypes.NewMsgSend(
+					vestAccAddr,
+					sendingAccAddr,
+					sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(1))),
+				)
+				return []sdk.Msg{msg}, vestAccName
+			},
+			expectedCode: abci.CodeTypeOK,
+		},
+	}
```

Comment:
one additional test we can do is to have a third test case where we attempt to send too much and expect it to fail. I have a commit with this ready, however I'm worried that it will cause flakiness do to the process requiring a specific time delay window.

Reply:
I don't think it will be flaky if you set the end time well in the future and try send all the funds

Reply:
> well in the future and try send all the funds

yeah good point, sorry I was not clear enough. the thing that I was trying to test with the additional case here was not just that a tx would fail if there was not enough of a spendable balance, I think we have that somehwere already (#2004 iirc). That wouldn't be flakey. 

I was trying to create a test that checks that puts us in the exact scenario we would get in if we used `time.Now()`. Where a tx doesn't get filtered but is invalid.